app/controllers/buyers/applications_controller.rb

```diff
@@ -1,236 +1,43 @@
 
   before_action :authorize_partners
-  before_action :find_buyer, only: %i[index create new]
-  before_action :find_plans, only: %i[new create]
-  before_action :authorize_multiple_applications, only: [:create]
-
-  before_action :find_cinstance, :except => [:index, :create, :new]
-  before_action :find_provider,  only: %i[new create update]
-
-  activate_menu :buyers
-
-  layout 'provider'
-
-  def index
-    # TODO: Editing this action may require editing Api::ApplicationsController
+  before_action :find_plans
+  before_action :find_buyer
+  before_action :authorize_multiple_applications, only: :create
+  before_action :find_application_plan, only: :create
+  before_action :find_service, only: :create
+  before_action :find_service_plan, only: :create
 
-    activate_menu :audience, :applications, :listing
-
-    @states = Cinstance.allowed_states.collect(&:to_s).sort
-    @search = ThreeScale::Search.new(params[:search] || params)
-    @application_plans = accessible_plans
-
-    if params[:service_id]
-      @service = accessible_services.find params[:service_id]
-      @search.service_id = @service.id
-    end
-
-    if params[:application_plan_id]
-      @plan = @application_plans.find params[:application_plan_id]
-      @search.plan_id = @plan.id
-      @service ||= @plan.service
-    end
-
-    if params[:account_id]
-      @search.account = @account.id
-      activate_menu :buyers, :accounts, :listing
-    end
-
-    @cinstances = accessible_not_bought_cinstances
-      .scope_search(@search).order_by(params[:sort], params[:direction])
-      .preload(:service, user_account: [:admin_user], plan: [:pricing_rules])
-      .paginate(pagination_params)
-      .decorate
+  helper_method :accessible_services
 
-    display_view_portion!(:service) if current_account.multiservice?
-  end
+  activate_menu :buyers, :accounts, :listing
 
   def new
-    @products = accessible_services
-
-    return activate_menu :audience, :applications, :listing if applications_context?
-
     @cinstance = @account.bought_cinstances.build
     extend_cinstance_for_new_plan
-
-    activate_menu :buyers, :accounts, :listing
   end
 
-  # TODO: this should be done by buy! method
+  # Create is handled by each individual controller so that it don't render a different 'new' template in case of an error
   def create
-    application_plan = @plans.find plan_id
-    service_plan = if service_plan_id = params[:cinstance].delete(:service_plan_id)
-                     application_plan.service.service_plans.find(service_plan_id)
-                   end
-
-    @cinstance = current_account.provider_builds_application_for(@account, application_plan, params[:cinstance], service_plan)
-    @cinstance.validate_human_edition!
+    super
 
     if @cinstance.save
       flash[:notice] = 'Application was successfully created.'
-      redirect_to(admin_service_application_path(@cinstance.service, @cinstance))
+      redirect_to provider_admin_application_path(@cinstance)
```

Comment:
I think it should redirect to the index page in the same scope (buyer, in this case).